WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -81,12 +115,25 @@ class Api::V0::UserRolesController < Api::V0::ApiController
 
   # Returns a list of roles by user which are not yet migrated to the new system.
   private def group_roles_not_yet_in_new_system(group_id)
-    group = UserGroup.find(group_id)
     roles = []
+    if group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
+      # migrated to the new system.
+      group_type = group_id.split("_").reverse.drop(1).reverse.join("_")
```

Comment:
This "reverse drop reverse join" looks very unintuitive and is repeated a lot in other methods further down.
Can you extract it into a helper method along with a comment explaning what it actually does?